testing/citest/tests/google_http_lb_upsert_scenario.py

```diff
@@ -480,7 +480,17 @@ def update_port_range(self):
   def add_cert(self, certname, title):
     '''Add cert to targetHttpProxy to make it a targetHttpsProxy.
     '''
+    # Note: We add this health check to the upsert description because
+    # this health check would be orphaned otherwise. This allows the 'delete'
+    # test to delete all the resources we allocated.
```

Comment:
```python
  def test_c_upsert_min_lb(self): 
    self.run_test_case(self.scenario.upsert_min_load_balancer()) # => Use proto health check for all backend services.

  def test_d_upsert_full_lb(self):
    self.run_test_case(self.scenario.upsert_full_load_balancer()) # => Use proto health check for all backend services.

  def test_e_add_security_group(self):
    self.run_test_case(self.scenario.add_security_group())

  # Test letters f and n are reserved for another test derived from this one.

  def test_g_change_hc(self):
    self.run_test_case(self.scenario.change_health_check()) # => Create new, updated health check and use for all backend services. Old proto health check still exists in GCP and is orphaned.

  def test_h_change_bs(self):
    self.run_test_case(self.scenario.change_backend_service()) # => Re-use proto health check for all backend services. Now the updated health check is orphaned.

  def test_i_add_host_rule(self):
    self.run_test_case(self.scenario.add_host_rule()) # => Re-use proto health check for all backend services. Now the updated health check is orphaned.

  def test_j_update_host_rule(self):
    self.run_test_case(self.scenario.update_host_rule()) # => Re-use proto health check for all backend services. Now the updated health check is orphaned.

  def test_k_update_port_range(self):
    self.run_test_case(self.scenario.update_port_range()) # => Re-use proto health check for all backend services. Now the updated health check is orphaned.

  def test_l_add_cert(self):
    self.run_test_case(self.scenario.add_cert(self.__class__.FIRST_CERT, 'add cert')) # => Add the updated health check to just one of the backend services, so both the updated health check and the proto health check are included in the load balancer.

def test_m_change_cert(self):
    self.run_test_case(self.scenario.add_cert(self.__class__.SECOND_CERT,
                                              'update cert')) # => Add the updated health check to just one of the backend services, so both the updated health check and the proto health check are included in the load balancer.

  # Test letters f and n are reserved for another test derived from this one.

  def test_o_delete_lb(self):
    self.run_test_case(self.scenario.delete_http_load_balancer()) # => Delete all components currently associated with the load balancer.
 
```

In one of the upsert operations, we create a _new_ health check that is an 'updated' health check to the L7 load balancer. Because we replace _all_ instances of the old health check in the L7 load balancer, the old (and thus unused) health check is now orphaned in GCP.

We then switch back to using the original health check for all the health checks in the L7 for a few upserts, which then makes the 'updated' health check orphaned in GCP. Before this PR, we would no longer use the updated health check in any of the rest of the operations so it would leak. This PR now makes it so that we add the 'updated' health check back into the load balancer so we can delete it.